Upgrade to LGDS 9.0 to reset prose styling

aduth opened the pull request:
## 🛠 Summary of changes

Updates to `@18f/identity-design-system@9.0.0-beta.2` (currently prerelease), which includes changes to reset prose styling. The initial implementation of this site included custom variant of the prose styles, presumably to work around some of the undesirable implications of the original styles.

As a draft, the intent is to assess potential impact and whether additional changes are necessary to the design system revisions before a final 9.0.0 release (as described at https://github.com/18F/identity-design-system/pull/390#issuecomment-1883779930).

This also reconfigures line-height to use a USWDS configurable value `1.75` value rather than an ad-hoc `1.875rem`, which may result in a slight reduction in line height.

## 📜 Testing Plan

- Go to https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/
- Check that "prose" content appears as expected:
   - ["What is Login.gov" page content](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/what-is-login/)
   - Any help article's content ([example](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/help/trouble-signing-in/how-to-sign-in/))
   - Any "sidenav" layout page content:
      - Any "Policy" section page content ([example](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/policy/))
      - ["Contact Us" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/contact/)
      - ["Join us" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/join/)
      - ["About us" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/about-us/)
      - ["Accessibility" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/accessibility/)

nickttng commented:
Hey, @aduth - I've noted several quirks in the table chart. The impact of prose styling becomes more apparent on pages that aren't long-form content like Policy or Help Center articles. In the Policy and Help Center articles, the prose styling update is minimal but does enhance, so these look good to me. Happy to discuss more in this issue or meet if it's easier. 

| Reference | Current | Prose | What's changed | Impact |
|---|---|---|---|--|
|Hero section|![Screenshot 2024-02-08 at 2 10 16 PM](https://github.com/18F/identity-site/assets/6327082/ff72b571-afb3-4377-a8f1-4cfd0a5532b0)|![Screenshot 2024-02-08 at 2 10 23 PM](https://github.com/18F/identity-site/assets/6327082/1d03460c-916a-47c8-9a0d-2d5c6ad6765a)|![Screenshot 2024-02-08 at 3 10 19 PM](https://github.com/18F/identity-site/assets/6327082/5b547051-deec-40a0-9da3-40748ea83cfd)|The hero description shows a noticeable line height increase; however, I'm okay with it. How does it look to you?|
|Marketing page|![Screenshot 2024-02-08 at 2 09 43 PM](https://github.com/18F/identity-site/assets/6327082/4aca8170-14a8-4a7e-b86c-cc9750268a79)|![Screenshot 2024-02-08 at 2 09 52 PM](https://github.com/18F/identity-site/assets/6327082/a7b373c9-1507-482b-8f46-b1012071a178)|![Screenshot 2024-02-08 at 2 21 55 PM](https://github.com/18F/identity-site/assets/6327082/e14086b8-a55e-4abc-9aa8-7a4f971f2ff3)|The prose styling introduces top and bottom margins to the heading. The visual impact is minimal on simple, long-form page layout; however, the effect becomes evident on the marketing pages, e.g., "What is Login.gov." The margins create extra white space, pushing other content further down. We should address that.|
|Contact us (Desktop)|![Screenshot 2024-02-08 at 2 12 09 PM](https://github.com/18F/identity-site/assets/6327082/d546918b-55b5-4fa8-874f-2a30f7675f1c)|![Screenshot 2024-02-08 at 2 12 02 PM](https://github.com/18F/identity-site/assets/6327082/cb745b8a-dd6b-41b3-85cf-c09ad7ba82ef)|![Screenshot 2024-02-08 at 3 16 32 PM](https://github.com/18F/identity-site/assets/6327082/5e37b1f0-45e6-4311-94ea-e36368e6da48)|Similar to the above, the additional margins push the content down.|
|Contact us (Mobile)|![login gov_contact_(iPhone SE)](https://github.com/18F/identity-site/assets/6327082/b3b319d6-52a6-4d39-a088-df1a0c8bdbf5)|![federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5 sites pages cloud gov_preview_18f_identity-site_aduth-rm-prose-styles_contact_(iPhone SE)](https://github.com/18F/identity-site/assets/6327082/46c31bc6-8351-404e-a8ab-461eddd0ae09)|![Screenshot 2024-02-08 at 3 23 38 PM](https://github.com/18F/identity-site/assets/6327082/8ebae729-23da-4891-bf08-172882387af7)|On mobile, the prose styling increases the line height spacing, eliminating the margin-top seen on the current site. This effect is more apparent on the "Contact Us" page than on other pages. We might want to address it.|

aduth commented:
@nickttng Thanks for taking a detailed look!

>The hero description shows a noticeable line height increase; however, I'm okay with it. How does it look to you?

Yeah, I think this would be related to the note in the original comment about "use a USWDS configurable value 1.75 value rather than an ad-hoc 1.875rem", though in practice I see it's actually `1.8` here, not `1.75`. In some cases it might result in a narrower line-height, since it's a unitless proportion (`1.8` times the font size, rather than specifically `1.875rem` = 30px), so in larger text like the hero description it's bigger than before. Personally I think that's how we should want line height to work, to scale relative to the font size?

>The prose styling introduces top and bottom margins to the heading. The visual impact is minimal on simple, long-form page layout; however, the effect becomes evident on the marketing pages, e.g., "What is Login.gov." The margins create extra white space, pushing other content further down. We should address that.

>Similar to the above, the additional margins push the content down.

I think part of the issue here is that Prose component should be the immediate wrapping element for content sections, since it applies many of its styles using [CSS child selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator). Since the substitution to use `usa-prose` here was applied on an ancestor element, the styles aren't quite correct. I'll review the pages where `usa-prose` exists to make sure it's applied on the immediate parent of the content, and see if that helps with these sorts of issues.

>On mobile, the prose styling increases the line height spacing, eliminating the margin-top seen on the current site. This effect is more apparent on the "Contact Us" page than on other pages. We might want to address it.

This is another quirk of how prose heading styles are applied with [CSS next-sibling selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Next-sibling_combinator), adding margin because there's a hidden alert banner on the page that's shown during maintenance windows (see screenshot in #1023). Ideally USWDS could account for hidden content, but I'll see if I can come up with another workaround. Typically if the heading were the first content, it wouldn't have a top margin.

aduth commented:
@nickttng I pushed up a few changes which should hopefully resolve some of the issues you identified:

- 4797b38 will apply prose styling to the correct wrapping elements on the ["What is Login.gov?" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/what-is-login/) to reduce some of the excess vertical margins, especially in headings
- 7285229 will address the issue with the excess top margin for the first heading on the ["Contact Us" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/contact/)
- 796e5c4 will address additional excess vertical margins on the ["Contact Us" page](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/contact/)

nickttng commented:
Thank you for these. I'll take another detailed look in the next few days

nickttng commented:
@aduth The changes look good. 

One last question: Do you have a stance on having a max-width for list items?

In this screenshot, the list width in the preview (right side) extends further than the list width on the current site (left side).

![Screenshot 2024-02-20 at 2 33 31 PM](https://github.com/18F/identity-site/assets/6327082/f02e63ce-8d02-43a6-b28b-a241e37286c2)

Several lines have a number of characters with more than 80 characters per [WCAG SC 1.4.8](https://www.w3.org/TR/WCAG22/#visual-presentation)

> Width is no more than 80 characters or glyphs (40 if CJK).

Is it possible to set a width with 80 characters or less per line?

aduth commented:
@nickttng That's a good catch. It looks like early on in this pull request I overrode the `$theme-text-measure` variable, likely to preserve [a previous `max-width: none`](https://github.com/18F/identity-site/blob/62555ac3373bfbb8730dc4cd79c4d1452c8f9a5b/_sass/components/_typography.scss#L35-L36) that had existed with our older ad hoc styling, but I don't think the `max-width` was doing what we thought it was there, since [USWDS applies `max-width` on the list item](https://github.com/uswds/uswds/blob/202448875393d39d663614d2d8f9a12c91fed75e/packages/uswds-core/src/styles/mixins/typography/usa-list.scss#L22), not on the list itself. In any case, [removing the override](https://github.com/18F/identity-site/pull/1207/commits/7f5d88ce3ae375032988328fc7c57f04967948e4) restores the default text measure, which should enforce a maximum width on list items and paragraphs within prose content.

nickttng commented:
> I assume we might actually want that, though?

@aduth Yes, I think so. Is the impact on paragraphs already reflected in the current build? (I think so, but want to double check.)

aduth commented:
Yep, it's already present in the preview. 

For example, you can see the content is slightly narrower on policy pages between the preview and production.

[Production](https://www.login.gov/policy/)|[Preview](https://federalist-17bd62cc-77b7-4687-9c62-39b462ce6fd5.sites.pages.cloud.gov/preview/18f/identity-site/aduth-rm-prose-styles/policy/)
---|---
![image](https://github.com/18F/identity-site/assets/1779930/1a2886f3-b62f-44f4-93d9-587a8402a42d)|![image](https://github.com/18F/identity-site/assets/1779930/37aac8f0-6714-470f-8400-494c5a055322)

nickttng commented:
The first line of the narrower version has a character count of `69` while the production version has a count of `98`. So I prefer the narrower version. How about you?

Also, any concern with the additional white spacing between the main content and the side navigation?

aduth commented:
>The first line of the narrower version has a character count of 69 while the production version has a count of 98. So I prefer the narrower version. How about you?

@nickttng I think I prefer the narrower versions as being consistent with our help articles, being conformant with WCAG 1.4.8, and aligning to USWDS defaults.

>Also, any concern with the additional white spacing between the main content and the side navigation?

Not especially concerned, especially since this is now consistent with the amount of whitespace that already exists on help articles that have a similar side navigation.

If we did have concerns, I'd wonder if we should (separately) reevaluate if the side navigation ought to be shown on the left instead of on the right, similar to what we're doing on the developer site redesign ([example](https://developers.login.gov/saml/getting-started/)).

nickttng commented:
Narrower version, then. 👍🏼 

> I'd wonder if we should (separately) reevaluate if the side navigation ought to be shown on the left instead of on the right

I agree. It'd be helpful to have the side nav on the left to better and quickly support users with orientation of site/page whereabout. Should we treat this as a separate work/topic to discuss what it involves in making it happen?

aduth commented:
>Should we treat this as a separate work/topic to discuss what it involves in making it happen?

Yep, that makes sense to me 👍

nickttng reviewed (approved):
Appreciate your work here!

aduth commented:
@nickttng FYSA I've updated this with the final release of LGDS 9.0. I added [a small revision](https://github.com/GSA-TTS/identity-site/pull/1207/commits/b088060f02ae90975bde5906b3f6b0293e2ce1ea) to icon lists in the Accepted Identification Documents pages after noticing the default USWDS styling removes margins from icon lists in prose content, which was causing it to bump very close to the alert on those pages.

I'll also plan to seek a technical review here.

zachmargolis commented on _includes/components/2-col.html:
```diff
@@ -1,5 +1,5 @@
-<div class="container container--mod {{ include.class }} page-content__prose">
-  <div class="one-account page-content__prose tablet:grid-row">
+<div class="container container--mod {{ include.class }}">
```
would it be worth documenting include params while we're we're here?
```suggestion
{% comment %}
includes
- class
- col1
- bannertext
{% endcomment %}
<div class="container container--mod {{ include.class }}">
```

aduth replied:
> would it be worth documenting include params while we're we're here?

Makes sense! Although honestly this layout is very misleading in presenting itself as something that can be reused, since it [hard-codes images from the "What is Login.gov?" page](https://login.gov/what-is-login/) 😬 

https://github.com/GSA-TTS/identity-site/blob/6d708f004c965e307c3bc7bf4c1b1858663e6abb/_includes/components/2-col.html#L8

https://github.com/GSA-TTS/identity-site/blob/6d708f004c965e307c3bc7bf4c1b1858663e6abb/_includes/components/2-col.html#L25

https://github.com/GSA-TTS/identity-site/blob/6d708f004c965e307c3bc7bf4c1b1858663e6abb/_includes/components/2-col.html#L34

https://github.com/GSA-TTS/identity-site/blob/6d708f004c965e307c3bc7bf4c1b1858663e6abb/_includes/components/2-col.html#L46

zachmargolis replied:
oof

aduth replied:
Updated to add comment in 7429bd7.

zachmargolis commented on _sass/components/_svg.scss:
```diff
@@ -3,8 +3,8 @@
   svg {
-    height: $icon-width;
-    width: $icon-width;
+    height: 0.8rem;
+    width: 0.8rem;
```
should this be derived from the font family like the line heights? or is it a separate calculation?

aduth replied:
I don't think I put much thought into this other than duplicating the existing value previously assigned, since it started erroring because USWDS now has a variable with the same name.

I'll poke at it and see what it's used for and how we'd want to deal with it.

aduth replied:
Removing the style separately in #1221

zachmargolis commented on assets/js/contact_us_form_element.ts:
```diff
@@ -39,6 +39,8 @@ class ContactUsFormElement extends HTMLElement {
     if (this.isInMaintenanceWindow) {
       this.showAlert();
       this.hide();
+    } else {
+      this.removeAlert();
     }
```
this is not prose style updating, this is USWDS API updates righjt?

aduth replied:
This was a [workaround](https://github.com/GSA-TTS/identity-site/pull/1207#issuecomment-1936113605) to a [visual regression on the contact page](https://github.com/GSA-TTS/identity-site/pull/1207#issuecomment-1934972525) with the introduction of prose, since the use of sibling selectors in how prose creates margins doesn't account for hidden content. So rather than relying on it staying hidden, this removes it from the page to avoid the weirdness with the CSS sibling selectors.

zachmargolis reviewed (approved):
LGTM

aduth commented:
I fixed up a couple more items in some final QA I ran over the preview. Also rebased to simplify slightly with the changes in #1221 being merged.

1. "Back to top" links lost their link styling, fixed in d07b28a

Before|After
---|---
![Screenshot 2024-03-08 at 8 10 12 AM](https://github.com/GSA-TTS/identity-site/assets/1779930/5836f249-9d37-48e1-8178-7eaeb54fde0b)|![Screenshot 2024-03-08 at 8 10 14 AM](https://github.com/GSA-TTS/identity-site/assets/1779930/5b4943ec-60ad-4ae1-95ef-0632d444e306)

2. Some content that had deeply embedded HTML links similarly lost their link styling, fixed in 6eefc04

Before|After
---|---
![Screenshot 2024-03-08 at 8 18 32 AM](https://github.com/GSA-TTS/identity-site/assets/1779930/f953ba31-e0c8-4420-bcfc-1b2acff4e371)|![Screenshot 2024-03-08 at 8 19 05 AM](https://github.com/GSA-TTS/identity-site/assets/1779930/5ab8716b-f9a9-459f-9608-da3b4b7229da)

I also noticed that the reduction of the content width doesn't apply to non-simple content such as alert banners, which can stand out a bit. @nickttng not sure how you feel about this, and maybe it's something we can follow-up separately. We could enforce a maximum width here as well, though it seems like something we may want to do from the design system, and perhaps more holistically than just one or two components?

Example:

![Screenshot 2024-03-08 at 8 24 09 AM](https://github.com/GSA-TTS/identity-site/assets/1779930/264920fc-70ed-4d91-9fd5-41fd632970e4)

zachmargolis reviewed (approved):
LGTM, following up in a separate PR for max text width inside alerts seems fine to me

nickttng commented:
> maybe it's something we can follow-up separately. We could enforce a maximum width here as well, though it seems like something we may want to do from the design system, and perhaps more holistically than just one or two components?

@aduth +1 to zach's on following up separately. we can discuss about if or how to approach it holistically at our upcoming LGDS office hours.
